pkg/controller/controller.go

```diff
@@ -993,16 +1018,28 @@ func (p *csiProvisioner) getVolumeContentSource(ctx context.Context, claim *v1.P
 
 // getPVCSource verifies DataSource.Kind of type PersistentVolumeClaim, making sure that the requested PVC is available/ready
 // returns the VolumeContentSource for the requested PVC
-func (p *csiProvisioner) getPVCSource(ctx context.Context, claim *v1.PersistentVolumeClaim, sc *storagev1.StorageClass) (*csi.VolumeContentSource, error) {
-	sourcePVC, err := p.claimLister.PersistentVolumeClaims(claim.Namespace).Get(claim.Spec.DataSource.Name)
+func (p *csiProvisioner) getPVCSource(ctx context.Context, claim *v1.PersistentVolumeClaim, sc *storagev1.StorageClass, dataSource *v1.ObjectReference) (*csi.VolumeContentSource, error) {
+
+	var sourcePVC *v1.PersistentVolumeClaim
+	var err error
+	if len(dataSource.Namespace) > 0 {
+		if claim.Namespace != dataSource.Namespace {
+			if ok, err := p.IsGranted(ctx, claim); err != nil || !ok {
```

Comment:
Can `IsGranted` also be moved to `dataSource`?

Reply:
Is it possible. Fixed code.